### PR TITLE
Add gocritic to linters

### PR DESCRIPTION
**Pull request by ulyssessouza**

### CODE REVIEW

**Review by nicksieger — APPROVED** [review]
LGTM

**Review by laurazard — APPROVED** [review]
LGTM

File: loader/loader.go [hunk 1]
```diff
@@ -664,16 +673,16 @@ func resolveVolumePath(volume types.ServiceVolumeConfig, workingDir string, look
 }
 
 // TODO: make this more robust
-func expandUser(path string, lookupEnv template.Mapping) string {
-	if strings.HasPrefix(path, "~") {
+func expandUser(p string, lookupEnv template.Mapping) string {
```
Comment (glours) [inline loader/loader.go:676]: sorry but I definitely not a fan of changing `path` to `p` to make a linter happy, we lost to much meaning of the function argument in that case! 
Isn't possible the alias the import instead?

**Review by glours — COMMENTED** [review]
If I understand correctly one of the rule is to not use a package name as a variable one, I'm ok with that only if we don't lose the meaning of the variable in our code base, my2cents

File: loader/loader.go (hunk 1 above)
Comment (ulyssessouza) [reply loader/loader.go:676]: Sure!

Comment (ulyssessouza) [reply loader/loader.go]: Renamed the import to `paths`

File: loader/validate.go [hunk 2]
```diff
@@ -38,6 +38,12 @@ func checkConsistency(project *types.Project) error {
 			}
 		}
 
+		for dependedService := range s.DependsOn {
```
Comment (nicksieger) [inline loader/validate.go:41]: Threw in a freebie, eh? Normally I'd say separate PR but it's a small change so 👍

**Review by nicksieger — APPROVED** [review]
👍 ♻️

File: loader/validate.go (hunk 2 above)
Comment (ulyssessouza) [reply loader/validate.go:41]: Wow! Good catch! But I didn't add that 😮 
It's already on `master` https://github.com/compose-spec/compose-go/blob/master/loader/validate.go#L41-L45

Comment (ulyssessouza) [reply loader/validate.go]: I've just rebased and it seems to remove this section from the diff

**Pull request merged**